tiled/adapters/arrow.py

```diff
@@ -0,0 +1,329 @@
+        Returns
+        -------
+        The reader handle for specific partition.
+        """
+        if not Path(self._partition_paths[partition]).exists():
+            raise ValueError(f"partition {partition} has not been stored yet")
+        else:
+            return pyarrow.ipc.open_file(self._partition_paths[partition])
+
+    @property
+    def reader_handle_all(self) -> Iterator[pyarrow.RecordBatchFileReader]:
+        """
+        Function to initialize and return the reader handle.
+        Returns
+        -------
+        The reader handle.
+        """
+        for path in self._partition_paths:
+            if not Path(path).exists():
+                raise ValueError(f"path {path} has not been stored yet")
+            else:
+                with pyarrow.ipc.open_file(path) as reader:
+                    yield reader
+
+    def write_partition(
+        self,
+        data: Union[List[pyarrow.record_batch], pyarrow.record_batch, pandas.DataFrame],
+        partition: int,
+    ) -> None:
+        """
+        "Function to write the data into specific partition as arrow format."
+        Parameters
+        ----------
+        data : data to write into arrow file. Can be a list of record batch, or pandas dataframe.
+        partition: integer index of partition to be read.
+        Returns
+        -------
+        """
+        if isinstance(data, pandas.DataFrame):
+            table = pyarrow.Table.from_pandas(data)
+            batches = table.to_batches()
+        else:
+            if not isinstance(data, list):
+                batches = [data]
+            else:
+                batches = data
+
+        schema = batches[0].schema
+
+        uri = self._partition_paths[partition]
+
+        with pyarrow.ipc.new_file(uri, schema) as file_writer:
+            for batch in batches:
+                file_writer.write_batch(batch)
+
+    def write(
+        self,
+        data: Union[List[pyarrow.record_batch], pyarrow.record_batch, pandas.DataFrame],
+    ) -> None:
+        """
+        "Function to write the data as arrow format."
+        Parameters
+        ----------
+        data : data to write into arrow file. Can be a list of record batch, or pandas dataframe.
+        Returns
+        -------
+        """
+        if isinstance(data, pandas.DataFrame):
+            table = pyarrow.Table.from_pandas(data)
+            batches = table.to_batches()
+        else:
+            if not isinstance(data, list):
+                batches = [data]
+            else:
+                batches = data
+
+        schema = batches[0].schema
+
+        if self.structure().npartitions != 1:
+            raise NotImplementedError
+        uri = self._partition_paths[0]
+
+        with pyarrow.ipc.new_file(uri, schema) as file_writer:
+            for batch in data:
+                file_writer.write_batch(batch)
+
+    def read(self, *args: Any, **kwargs: Any) -> pandas.DataFrame:
```

Comment:
This currently accepts any positional or keyword arguments and then ignores them. Notice that in CSV, we accept any arguments but we _pass them into another method_ which accepts specific arguments and applies them:

https://github.com/bluesky/tiled/blob/15abbc98df9bfd25f8c713656f664a558136e26f/tiled/adapters/csv.py#L203-L217

It should accept an optional `fields` parameter and apply it if it is not `None`. https://github.com/bluesky/tiled/blob/15abbc98df9bfd25f8c713656f664a558136e26f/tiled/adapters/table.py#L226-L227